examples/hosts/app-integration/document-recovery/src/recoveryAgent.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import { ContainerSchema, SharedMap } from "fluid-framework";
+import {
+    AzureClient,
+    AzureContainerVersion,
+} from "@fluidframework/azure-client";
+import { connectionConfig } from "./azureConfig";
+import { counterValueKey } from "./dataController";
+
+export type RecoveryStatus =
+    | "NotStarted"
+    | "KickedOff"
+    | "Success"
+    | "Error"
+    | "Cancelled";
+
+export interface RecoveryInfo {
+    originalContainerId: string;
+    isContainerRecovered?: boolean;
+    recoveredContainerId?: string;
+    recoveryStatus?: RecoveryStatus;
+    recoveryLog: string;
+}
+
+export class RecoveryAgent extends EventEmitter {
+    private readonly azureClient: AzureClient;
+    private readonly maxDocVersions = 5;
+    private recoveryLog: string;
+    private recoveryStatus: RecoveryStatus;
+    public recoveredContainerId: string | undefined;
+
+    constructor(
+        private readonly orgContainerId: string,
+        private readonly orgContainerSchema: ContainerSchema,
+    ) {
+        super();
+        const clientProps = {
+            connection: connectionConfig,
+        };
+        this.azureClient = new AzureClient(clientProps);
+        this.recoveredContainerId = undefined;
+        this.recoveryLog = "";
+        this.recoveryStatus = "NotStarted";
+    }
+
+    public static createRecoveryAgent(
```

Comment:
why do we have a dedicated static function for creating a recovery agent rather than directly instantiating it with `RecoveryAgent(containerId, containerSchema)`

Reply:
This call had somewhat different purpose in earlier version of this example. Not needed here any further.